tests/nosetests/pyanaconda_tests/__init__.py

```diff
@@ -189,29 +188,36 @@ def check_task_creation_list(test, task_paths, publisher, task_classes):
 
     :return: list of instances of tasks
     """
-    res = []
+    task_proxies = []
+    task_count = len(task_paths)
 
-    for i in range(len(task_paths)):
-        res.append(check_task_creation(test, task_paths[i], publisher, task_classes[i]))
+    # Check the number of published tasks.
+    test.assertEqual(task_count, publisher.call_count)
+    test.assertEqual(task_count, len(task_classes))
 
-    return res
+    # Check each published task.
+    for i in range(task_count):
+        task_proxy = check_task_creation(test, task_paths[i], publisher, task_classes[i], i)
+        task_proxies.append(task_proxy)
 
+    return task_proxies
 
-def check_dbus_object_creation(test, path, publisher, klass):
+
+def check_dbus_object_creation(test, path, publisher, klass, index=0):
     """Check that the custom DBus object is correctly created.
 
     :param test: instance of TestCase
-    :param task: DBus path of the published object
+    :param path: DBus path of the published object
     :param publisher: Mock instance of the publish_object method
     :param klass: class of the tested DBus object
+    :param index: an index of the published object
     """
-    publisher.assert_called_once()
-    object_path, obj = publisher.call_args[0]
+    test.assertGreater(publisher.call_count, index)
```

Comment:
Could you please add explanation why the call count have to be greater than index? Is that because index starts from 0 but call count is counted from 1?

Reply:
Fixed.